backend/app/users/routers.py

```diff
@@ -0,0 +1,87 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+
+from ..auth.dependencies import authenticate_either
+from ..database import get_async_session
+from .models import Users
+from .schemas import UserCreate, UserOut
+from .utils import hash_password
+
+TAG_METADATA = {
+    "name": "User Management",
+    "description": "Endpoints for adding, deleting and modifying users.",
+}
+
+router = APIRouter(
+    dependencies=[Depends(authenticate_either)],
+    tags=["User Management"],
+    prefix="/users",
+)
+
+
+@router.post("/create", response_model=UserOut)
+async def create_user(
+    user_data: UserCreate,
+    session: AsyncSession = Depends(get_async_session),
+) -> UserOut:
+    """
+    Create a new user.
+    """
+    if user_data.role not in ["admin", "user", None]:
+        raise HTTPException(
+            status_code=400,
+            detail="Role must be either 'admin' or 'user' or left blank",
+        )
+    hashed_password = await hash_password(user_data.password)
+    new_user = Users(
+        email=user_data.email, role=user_data.role, password=hashed_password
+    )
+
+    session.add(new_user)
+    try:
+        await session.commit()
+    except IntegrityError:
```

Comment:
can we get the IntegrityError as `e` and then raise HTTPException from `e`? More detail is good!